src/components/TabsBar/TabsBar.vue

```diff
@@ -0,0 +1,216 @@
+<template>
+    <div id="tabsBar" class="noSelect pointerCursor">
+        <draggable
+            :key="updateCount"
+            v-model="SimulatorState.circuit_list"
+            class="list-group"
+            tag="transition-group"
+            :component-data="{
+                tag: 'div',
+                type: 'transition-group',
+                name: !drag ? 'flip-list' : null,
+            }"
+            v-bind="dragOptions"
+            @start="drag = true"
+            @end="drag = false"
+        >
+            <template #item="{ element }">
+                <div
+                    :id="element.id"
+                    :key="element.id"
+                    style=""
+                    class="circuits toolbarButton"
+                    :class="tabsbarClasses(element.id)"
+                    draggable="true"
+                    @click="switchCircuit(element.id)"
+                >
+                    <span class="circuitName noSelect">
+                        {{ truncateString(scopeList[element.id].name, 18) }}
+                    </span>
+                    <span
+                        id="scope.id"
+                        class="tabsCloseButton"
+                        @click="closeCircuit($event, element)"
+                    >
+                        x
+                    </span>
+                </div>
+            </template>
+        </draggable>
+        <button @click="createNewCircuit()">&#43;</button>
+    </div>
+    <MessageBox
+        v-model="SimulatorState.dialogBox.create_circuit"
+        :circuit-item="circuitToBeDeleted"
+        :button-list="buttonArr"
+        :input-list="inputArr"
+        :is-persistent="persistentShow"
+        :message-text="messageVal"
+        @button-click="
+            (selectedOption, circuitItem, circuitNameVal) =>
+                dialogBoxConformation(
+                    selectedOption,
+                    circuitItem,
+                    circuitNameVal
+                )
+        "
+    />
+</template>
+
+<script lang="ts" setup>
+import draggable from 'vuedraggable'
+import { showMessage, truncateString } from '#/simulator/src/utils'
+import { ref } from '@vue/reactivity'
+import { computed, onMounted, onUpdated } from '@vue/runtime-core'
+import {
+    createNewCircuitScope,
+    deleteCurrentCircuit,
+    getDependenciesList,
+    scopeList,
+    switchCircuit,
+} from '#/simulator/src/circuit'
+import MessageBox from '#/components/MessageBox/messageBox.vue'
+import { useState } from '#/store/SimulatorStore/state'
+
+const SimulatorState = useState()
```

Comment:
I can notice there are some wrong patterns on how we are using store here, lets have a call regarding this?